webapp/platform/design-system/src/components/primitives/tag/tag.tsx

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React, {forwardRef, memo, useMemo} from 'react';
+import type {ReactNode, MouseEventHandler} from 'react';
+import {useIntl} from 'react-intl';
+
+import WithTooltip from '../with_tooltip';
+
+import './tag.scss';
+
+export type TagSize = 'xs' | 'sm' | 'md' | 'lg';
+
+export type TagVariant =
+    | 'default'
+    | 'info'
+    | 'success'
+    | 'warning'
+    | 'danger'
+    | 'dangerDim'
+    | 'primary'
+    | 'secondary';
+
+export type TagPreset = 'beta' | 'bot' | 'guest' | 'custom';
+
+export interface TagProps {
+
+    /** The text content of the tag */
+    text?: ReactNode;
+
+    /** Predefined tag type (beta, bot, guest, or custom for manual text) */
+    preset?: TagPreset;
+
+    /** Size variant of the tag */
+    size?: TagSize;
+
+    /** Visual variant/color scheme of the tag */
+    variant?: TagVariant;
+
+    /** Whether to display text in uppercase */
+    uppercase?: boolean;
+
+    /** Icon element to display before the text (React component) */
+    icon?: ReactNode;
+
+    /** Icon size in pixels - will be auto-calculated based on tag size if not provided */
+    iconSize?: number;
+
+    /** Click handler for interactive tags */
+    onClick?: MouseEventHandler<HTMLElement>;
+
+    /** Additional CSS class names */
+    className?: string;
+
+    /** Test ID for testing purposes */
+    testId?: string;
+
+    /** Tooltip content to display on hover */
+    tooltip?: ReactNode;
+
+    /** Tooltip content (alternative prop name) */
+    tooltipTitle?: ReactNode;
+
+    /** Whether to hide the tag (useful for conditional rendering like guest tags) */
+    hide?: boolean;
+
+    /** Full width variant */
+    fullWidth?: boolean;
```

Comment:
Do we actually need this? I don't see any reference to this in the docs or anywhere that we use it

Reply:
I don't see a need for it. I say remove it